plugins/modules/host.py

```diff
@@ -201,69 +229,98 @@ def process(self):
                     host_id = host['host_id']
                     existing = host_api_instance.read_host(host_id=host_id).to_dict()
                     break
-                
+                elif self.host_ip == host['ip_address']:
+                    host_id = host['host_id']
+                    existing = host_api_instance.read_host(host_id=host_id).to_dict()
+                    break
         except ApiException as ex:
             if ex.status != 404:
                 raise ex  
             
-        if self.state in ['present']:
+        if self.state == 'present':
             if existing:
-                self.host_output = existing
+                host_id = existing['host_id']
             else:
-                self.check_required_host_id_param()
-                body = {"items": [{"hostname": self.host_name,"ipAddress": self.host_ip }]}
-                create_host = host_api_instance.create_hosts(body=body).to_dict()
-                self.host_output = host_api_instance.read_host(host_id=create_host['items'][0]['host_id']).to_dict()
-                self.changed = True
+                host_params = {
+                    "hostname": self.host_name,
+                    "ip_address": self.host_ip,
+                }
+                if self.rack_id:
+                    host_params["rack_id"] = self.rack_id
+                if not self.module.check_mode:
+                    host_list = ApiHostList(items=[ApiHost(**host_params)])
+                    create_host = host_api_instance.create_hosts(body=host_list).to_dict()
+                    host_id = create_host['items'][0]['host_id']
+                    self.changed = True
```

Comment:
`changed` should be set outside of the `check_mode` gate -- you want to be able to run `check_mode` and know what will be changed